docs/src/cookbooks/naming.md

```diff
@@ -72,12 +72,44 @@ ChiselStage.emitSystemVerilog(new ExampleWhenPrefix)
 `_GEN` signals are usually generated from the FIRRTL compiler, rather than the Chisel library. We are working on
 renaming these signals with more context-dependent names, but it is a work in progress. Thanks for caring!
 
-### My module names are super unstable - I change one thing and Queue_1 becomes Queue_42. Help!
+### My module names are super unstable - I change one thing and Module_1 becomes Module_42. Help!
 
-This is the infamous `Queue` instability problem. In general, these cases are best solved at the source - the module
-itself! If you overwrite `desiredName` to include parameter information (see the
-[explanation](../explanations/naming#set-a-module-name) for more info), then this can avoid this problem permanantly.
-We've done this with some Chisel utilities with great results!
+This is an example of the module instability problem, which results from several modules all sharing the exact same name. To fix this, you must add more specificity to your `Module`'s name to avoid these name collisions.
+
+This can be done by leveraging the `desiredName` and `typeName` APIs. 
+`desiredName` is for indicating the names of `Modules` (e.g. influenced by the parameters passed in), and `typeName` is useful for modules which are type-parameterized by subclasses of `Data`. Overriding `desiredName` can reduce or even eliminate name collisions. For instance, suppose your module looks like the following:
+
+```scala mdoc
+class MyModule[T <: Data](gen: T) extends Module {
+  val in = IO(Input(gen))
+  val out = IO(Output(gen))
+  out := in
+}
+```
+
+We can override `desiredName` to include the type name of the `gen` parameter like so:
+
+```scala mdoc
```

Comment:
I'd probably show this as a subclass of MyModule, or just write it out again (you can use a bunch of tags to reset the compilation context, or just give it a new name) , so that the mdoc will actually compile. If this is how you want it, you can just omit the mdoc tag.

Reply:
> you can use a bunch of tags to reset the compilation context

How would I do this? I'd like to show that the `MyModule` example can be supplemented with overriding `desiredName` to demonstrate use of the `typeName` API